pkgs/applications/blockchains/lighthouse/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config, cmake, protobuf
+, zlib, bzip2, openssl, leveldb
+, nodePackages, curl, gnumake, git
+}:
+
+let
+  # Contains the data required by the deposit_contract
+  # See `common/deposit_contract/build.rs` in the lighthous source for exact version
+  eth2specs = fetchFromGitHub {
+    owner = "ethereum";
+    repo = "eth2.0-specs";
+    rev = "v0.12.1";
+    sha256 = "05irxnskcsdbp844hgk2hwkjmgfjxin99jr1fl5b5wg1v7v7gl4y";
+  };
+  eth2specs_unsafe = fetchFromGitHub {
+    owner = "sigp";
+    repo = "unsafe-eth2-deposit-contract";
+    # NOTE: the version of the unsafe contract lags the main tag, but the v0.9.2.1 code is compatible
+    # with the unmodified v0.12.1 contract. See `common/deposit_contract/build.rs` for details.
+    rev = "v0.9.2.1";
+    sha256 = "151ahdj5114xb5hbhzsx5rx4yyry832hhm0bkx36s2np3r8v883q";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lighthouse";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "sigp";
+    repo = "lighthouse";
+    rev = "v${version}";
+    leaveDotGit = true; # Needed to get the proper version
```

Comment:
Can we inject the version easily? Leaving dotGit is kind of a last resort.

Reply:
I tried this with https://github.com/NixOS/nixpkgs/pull/106063/files#diff-d8165e975d221069554707389b93737b9aa76e9a4ac2b62e7f466bb8d3494bd5 but I need to find a better solution that I can submit upstream.

Leaving the dotGit doesn't even work for some reason. I will remove it in the next iteration.